src/engine/init.c

```diff
@@ -753,10 +754,9 @@ server_init(int argc, char *argv[])
 		D_GOTO(exit_mod_init, rc);
 	D_INFO("Network successfully initialized\n");
 
-	rc = daos_hhash_init();
+	rc = daos_hhash_init_feats(D_HASH_FT_RWLOCK);
```

Comment:
I almost forgot details, would "D_HASH_FT_RWLOCK | D_HASH_FT_LRU" be a little better?

Reply:
If D_HASH_FT_LRU is used, read will use write lock(because of moving caching entry..) I tested with/without LRU flag on the client side before, it looks disable LRU(make read-write lock usable) gives better performance....

Reply:
as discussed on teams, serverside use PTR type, so probably need not this PR.
see d_hhash_is_ptrtype related code.

Reply:
> RW lock is not much better when there are more than handful threads, does it require a lot of changes if we switch to per-thread hash table? Do we use the hash table for object handle or container handle? I can't remember....

for server-side the handle is just using pointer, bypassed hash-table but jut keep the same API as client.